commons-math/src/main/java/org/apache/commons/math4/optim/nonlinear/scalar/noderiv/BOBYQAOptimizer.java

```diff
@@ -0,0 +1,2475 @@
+                        currentBest.setEntry(i, lowerBound[i]);
+                    }
+                    if (trustRegionCenterOffset.getEntry(i) == upperDifference.getEntry(i)) {
+                        currentBest.setEntry(i, upperBound[i]);
+                    }
+                }
+                f = fAtInterpolationPoints.getEntry(trustRegionCenterInterpolationPointIndex);
+            }
+            return f;
+        }
+        default: {
+            throw new MathIllegalStateException(LocalizedFormats.SIMPLE_MESSAGE, "bobyqb");
+        }}}
+    } // bobyqb
+
+    // ----------------------------------------------------------------------------------------
+
+    /**
+     *     The arguments N, NPT, XPT, XOPT, BMAT, ZMAT, NDIM, SL and SU all have
+     *       the same meanings as the corresponding arguments of BOBYQB.
+     *     KOPT is the index of the optimal interpolation point.
+     *     KNEW is the index of the interpolation point that is going to be moved.
+     *     ADELT is the current trust region bound.
+     *     XNEW will be set to a suitable new position for the interpolation point
+     *       XPT(KNEW,.). Specifically, it satisfies the SL, SU and trust region
+     *       bounds and it should provide a large denominator in the next call of
+     *       UPDATE. The step XNEW-XOPT from XOPT is restricted to moves along the
+     *       straight lines through XOPT and another interpolation point.
+     *     XALT also provides a large value of the modulus of the KNEW-th Lagrange
+     *       function subject to the constraints that have been mentioned, its main
+     *       difference from XNEW being that XALT-XOPT is a constrained version of
+     *       the Cauchy step within the trust region. An exception is that XALT is
+     *       not calculated if all components of GLAG (see below) are zero.
+     *     ALPHA will be set to the KNEW-th diagonal element of the H matrix.
+     *     CAUCHY will be set to the square of the KNEW-th Lagrange function at
+     *       the step XALT-XOPT from XOPT for the vector XALT that is returned,
+     *       except that CAUCHY is set to zero if XALT is not calculated.
+     *     GLAG is a working space vector of length N for the gradient of the
+     *       KNEW-th Lagrange function at XOPT.
+     *     HCOL is a working space vector of length NPT for the second derivative
+     *       coefficients of the KNEW-th Lagrange function.
+     *     W is a working space vector of length 2N that is going to hold the
+     *       constrained Cauchy step from XOPT of the Lagrange function, followed
+     *       by the downhill version of XALT when the uphill step is calculated.
+     *
+     *     Set the first NPT components of W to the leading elements of the
+     *     KNEW-th column of the H matrix.
+     * @param knew
+     * @param adelt
+     */
+    private double[] altmov(
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 53. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1282-1565 into a separate method.